pkg/remoteimage/pull.go

```diff
@@ -39,11 +41,14 @@ func (p puller) ImageSize(ctx context.Context) int {
 	return int(imageStatusResponse.Image.Size_)
 }
 
-func (p puller) Pull(ctx context.Context) (err error) {
+func (p puller) Pull(ctx context.Context, pod string, ns string) (err error) {
 	repo := p.image.Name()
 	imageSpec := &cri.ImageSpec{Image: p.image.String()}
 	creds, withCredentials := p.keyring.Lookup(repo)
 	if !withCredentials {
+
+		p.checkForImageSize(nil, pod, ns)
```

Comment:
Would it be possible and helpful to add tests for getting image size with and without creds?